libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2245,6 +2245,22 @@ void TypeChecker::typeCheckFunctionGeneralChecks(
 		}
 	}
 
+	if (_functionType->kind() == FunctionType::Kind::BytesConcat)
```

Comment:
I'm wondering if it would be better to handle this similar to `abi.encode` and create a call to a new funciton inside the switch in `TypeChecker::visit(FunctionCall` and then call the "general checks" function from there.

Reply:
I'll try doing it like that now

Reply:
Done